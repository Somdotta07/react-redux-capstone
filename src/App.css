@import url('https://fonts.googleapis.com/css2?family=Lato&display=swap');

:root {
  --primary-color: rgb(223, 37, 74);
  --secondary-color: rgb(214, 126, 143);
  --place-color: #fff;
}

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: 'Lato', sans-serif;
}

nav {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  background-color: var(--primary-color);
  align-items: center;
}

.lessThan {
  display: flex;
  flex-direction: row;
  align-items: center;

  /* margin-top: 10px; */
}

.sign {
  margin-top: 5px;
}

h3 {
  margin-left: 40px;
}

.mic {
  margin-left: 80px;
}

.line-1 {
  color: var(--primary-color);
  overflow: hidden;
  border-right: 0.15em solid orangered;
  white-space: nowrap;
  letter-spacing: 0.15em;
  animation:
    typing 3.5s steps(30, end) infinite,
    cursor-blink 0.75s step-end infinite;
}

/* Animation */
@keyframes typing {
  from {
    width: 0;
  }

  to {
    width: 100%;
  }
}

/* The typewriter cursor effect */
@keyframes cursor-blink {
  from,
  to {
    border-color: transparent;
  }

  50% {
    border-color: orangered;
  }
}

.main-c {
  height: 100vh;
  background-image: url(./Common/corona-5090522_1280.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  margin-top: 10px;
}

.country-heading {
  height: 25vh;
  background-image: url(./Common/who_share.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  font-size: 0.8em;
  font-weight: bold;
  padding: 3px 10px;
  letter-spacing: 0.5px;
}

.countries-con {
  display: grid;
  grid-template-columns: 1fr 1fr;
  background-color: var(--secondary-color);
  padding: 10px;
  column-gap: 10px;
  row-gap: 10px;
  margin-top: 50px;
}

.country-name {
  height: 20vh;
  width: 45vw;
  background-color: var(--primary-color);
  background-image: url(./Common/imageedit_4_5887518972.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 5px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  transition: width 2s;
}

.country-name:hover {
  width: 300px;
}

.country-d {
  color: var(--place-color);
  font-size: 16px;
  font-weight: 800;
  text-decoration: none;
}

.recovered {
  color: var(--place-color);
  font-size: 0.8em;
  overflow: hidden;
}

.recovered > div > div {
  color: #fff;
  padding: 4px 12px;
  height: 45px;
  margin-bottom: 45px;
  display: inline-block;
}

.recovered div:first-child {
  animation: show 5s linear infinite;
}

@keyframes show {
  0% { margin-top: -270px; }
  5% { margin-top: -180px; }
  33% { margin-top: -180px; }
  38% { margin-top: -90px; }
  66% { margin-top: -90px; }
  71% { margin-top: 0; }
  99.99% { margin-top: 0; }
  100% { margin-top: -270px; }
}

#input-country {
  width: 100%;
  padding: 5px;
  padding-right: 40px;
  outline: none;
  border: 2px solid #999;
  border-radius: 5px;
  background-color: #fbfbfb;
  font-family: Cambria, Cochin, Georgia, serif;
  font-size: 16px;
  background-position: right;
  background-image: url('./Common/search.png');
  background-size: 30px;
  background-repeat: no-repeat;
}

.title {
  display: block;
  background-color: var(--primary-color);
  color: #fff;
  font-weight: bold;
  padding: 3px 10px;
  letter-spacing: 0.5px;
  font-size: 0.8em;
}

.data {
  display: flex;
  justify-content: space-between;
  background-color: var(--secondary-color);
  margin: 5px;
  padding: 5px;
  border-radius: 5px;
  color: #fff;
  font-weight: bold;
}

.details-country {
  height: 20vh;
  display: flex;
  justify-content: center;
  background-color: var(--secondary-color);
  background-image: url(./Common/imageedit_1_7542201164.jpg);
  filter: blur(8px);
  -webkit-filter: blur(8px);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  margin: 8px;
  padding: 10px;
  border-radius: 5px;
  color: #fff;
  font-weight: bold;
}

.name {
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);
  color: rgb(234, 241, 227);
  border: 5px solid #f1f1f1;
  animation: growth 3s;
  animation-fill-mode: backwards;
  position: absolute;
  top: 18%;
  left: 50%;
  transform: translate(-50%, -60%);
  z-index: 2;
  width: 80%;
  padding: 10px;
  text-align: center;
  font-size: 16px;
  font-weight: 700;
  text-shadow: 2px 2px var(--primary-color);
  overflow-x: hidden;
}

@keyframes growth {
  0% {
    font-size: 0;
  }

  100% {
    font-size: 10px;
  }
}