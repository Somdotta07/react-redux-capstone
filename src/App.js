import { React } from 'react';
import './App.css';
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from 'react-router-dom';
import HomePage from './Components/HomePage';
import DetailsPage from './Components/DetailsPage';
import Header from './Components/Header';

function App() {
  return (
    <Router>
      <header>
        <Header />
      </header>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/details" element={<DetailsPage />} />
      </Routes>
    </Router>
  );
}

export default App;
