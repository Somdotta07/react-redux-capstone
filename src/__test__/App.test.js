import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import App from '../App';

describe('App page is working fine', () => {
  test('renders App', () => {
    const tree = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
