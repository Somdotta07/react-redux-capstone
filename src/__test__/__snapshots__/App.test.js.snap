// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page is working fine renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="header"
        >
          <nav
            class="nav"
          >
            <div
              class="lessThan"
            >
              <a
                href="/"
              >
                <svg
                  class="sign"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M365.46 357.74L147.04 255.89l218.47-101.88c16.02-7.47 22.95-26.51 15.48-42.53l-13.52-29C360 66.46 340.96 59.53 324.94 67L18.48 209.91a32.014 32.014 0 0 0-18.48 29v34.24c0 12.44 7.21 23.75 18.48 29l306.31 142.83c16.06 7.49 35.15.54 42.64-15.52l13.56-29.08c7.49-16.06.54-35.15-15.53-42.64z"
                  />
                </svg>
              </a>
              <h1
                class="logo"
              >
                2021
              </h1>
            </div>
            <h3>
              Covid Data
            </h3>
            <div
              class="mic"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 352 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
                />
              </svg>
            </div>
            <div
              class="settings"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
                <path
                  d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                />
              </svg>
            </div>
          </nav>
        </div>
      </header>
      <div
        class="main-c"
      >
        <div
          class="country-heading"
        >
          <h2
            class="line-1 anim-typewriter"
          >
            Active Corona Cases 
          </h2>
        </div>
        <input
          class="search"
          id="input-country"
          placeholder="Country name.."
          type="text"
        />
        <div
          class="countries-con"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="header"
      >
        <nav
          class="nav"
        >
          <div
            class="lessThan"
          >
            <a
              href="/"
            >
              <svg
                class="sign"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M365.46 357.74L147.04 255.89l218.47-101.88c16.02-7.47 22.95-26.51 15.48-42.53l-13.52-29C360 66.46 340.96 59.53 324.94 67L18.48 209.91a32.014 32.014 0 0 0-18.48 29v34.24c0 12.44 7.21 23.75 18.48 29l306.31 142.83c16.06 7.49 35.15.54 42.64-15.52l13.56-29.08c7.49-16.06.54-35.15-15.53-42.64z"
                />
              </svg>
            </a>
            <h1
              class="logo"
            >
              2021
            </h1>
          </div>
          <h3>
            Covid Data
          </h3>
          <div
            class="mic"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 352 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
              />
            </svg>
          </div>
          <div
            class="settings"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3"
              />
              <path
                d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
              />
            </svg>
          </div>
        </nav>
      </div>
    </header>
    <div
      class="main-c"
    >
      <div
        class="country-heading"
      >
        <h2
          class="line-1 anim-typewriter"
        >
          Active Corona Cases 
        </h2>
      </div>
      <input
        class="search"
        id="input-country"
        placeholder="Country name.."
        type="text"
      />
      <div
        class="countries-con"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
